frame/composable-support/src/collections/vec/bounded/sorted_vec.rs

```diff
@@ -0,0 +1,344 @@
+use crate::collections::vec::sorted::SortedVec;
+use codec::{Decode, Encode, EncodeLike, MaxEncodedLen};
+use core::{
+	ops::{Deref, Index},
+	slice::SliceIndex,
+};
+use frame_support::{traits::Get, BoundedVec};
+use sp_std::{convert::TryFrom, fmt, marker::PhantomData, prelude::*};
+
+/// A bounded, sorted vector.
+///
+/// It has implementations for efficient append and length decoding, as with a normal `Vec<_>`, once
+/// put into storage as a raw value, map or double-map.
+///
+/// As the name suggests, the length of the queue is always bounded and sorted. All internal
+/// operations ensure this bound is respected and order is maintained.
+#[derive(Encode)]
+pub struct BoundedSortedVec<T: Ord, S>(SortedVec<T>, PhantomData<S>);
+
+impl<T: Decode + Ord, S: Get<u32>> Decode for BoundedSortedVec<T, S> {
+	#[inline]
+	fn decode<I: codec::Input>(input: &mut I) -> Result<Self, codec::Error> {
+		let inner = Vec::<T>::decode(input)?;
+		if inner.len() > S::get() as usize {
+			return Err("BoundedSortedVec exceeds its limit".into())
+		}
+		Ok(Self(SortedVec::from_unsorted(inner), PhantomData))
```

Comment:
hm, feels risky to add O(n log N) into serde.

Reply:
how to express weight in such case?  seem need to write custom weight depending on length some how not only linearly. example could be use full If n log n is case

Reply:
Oh yeah correct, i'll instead check if sorted and otherwise error out